compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -572,6 +529,16 @@ fn link_staticlib<'a>(
         )
         .unwrap();
 
+        let bundled: FxHashSet<_> = native_libs.iter().filter_map(|lib| lib.filename).collect();
+        archive_builder_builder
+            .extract_bundled_libs(path, tempdir.as_ref(), &bundled)
+            .unwrap_or_else(|e| sess.emit_fatal(e));
+        for name in bundled {
+            let joined = tempdir.as_ref().join(name.as_str());
+            let path = joined.as_path();
+            ab.add_archive(path, Box::new(|_| false)).unwrap();
+        }
```

Comment:
This code is new.
Does it mean that `fn link_staticlib` didn't correctly support packed native libs previously?

Reply:
It seems so. I couldn't find code that correctly treat packed native libraries to generate staticlib.